CMakeLists.txt

```diff
@@ -395,8 +395,3 @@ add_library(SDL2::ttf INTERFACE IMPORTED GLOBAL)
 set_target_properties(SDL2::ttf PROPERTIES
     INTERFACE_LINK_LIBRARIES "SDL2_ttf"
 )
-if(CMAKE_VERSION VERSION_GREATER_EQUAL "3.17")
```

Comment:
This `if` conditional is supposed to protect against cmake versions < 3.17.
Please extract the [following file](https://github.com/libsdl-org/SDL_ttf/files/9711318/a.zip) to e.g. `/tmp/a.cmake`,
run it using `cmake -P /tmp/a.cmake` and post the output.

script:
```cmake
message("CMAKE_VERSION: ${CMAKE_VERSION}")
if(CMAKE_VERSION VERSION_GREATER_EQUAL "3.17.0")
  message("CMAKE_VERSION VERSION_GREATER_EQUAL 3.17: YES")
else()
  message("CMAKE_VERSION VERSION_GREATER_EQUAL: NO")
endif()
if(NOT CMAKE_VERSION VERSION_LESS "3.17.0")
  message("NOT CMAKE_VERSION VERSION_LESS 3.17: YES")
else()
  message("NOT CMAKE_VERSION VERSION_LESS 3.17: NO")
endif()
```

For my system, the output is:
```sh
$ cmake -P /tmp/a.cmake
CMAKE_VERSION: 3.22.2
CMAKE_VERSION VERSION_GREATER_EQUAL 3.17: YES
NOT CMAKE_VERSION VERSION_LESS 3.17: YES
```

Reply:
for my system was that
`CMAKE_VERSION: 3.16.3
CMAKE_VERSION VERSION_GREATER_EQUAL: NO
NOT CMAKE_VERSION VERSION_LESS 3.17: NO`

But the problem was in the 
`set_target_properties` that use DEPRECATED and was not allowed.

Reply:
> for my system was that `CMAKE_VERSION: 3.16.3 CMAKE_VERSION VERSION_GREATER_EQUAL: NO NOT CMAKE_VERSION VERSION_LESS 3.17: NO`

This output is correct for CMake 3.16.3.

> But the problem was in the `set_target_properties` that use DEPRECATED and was not allowed.

The [`DEPRECATION` property](https://cmake.org/cmake/help/v3.17/prop_tgt/DEPRECATION.html) has been available since CMake 3.17.
So the `set_target_properties` line should not have been reached.

I tried [CMake 3.16.3 for Linux](https://cmake.org/files/v3.16/cmake-3.16.3-Linux-x86_64.tar.gz) and the last lines of configuring this project with `cmake --trace` are:
```
/home/maarten/programming/SDL_ttf/CMakeLists.txt(398):  if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.17 )
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/build
```
==> The `set_target_properties` inside the `if` branch is skipped.